common/src/main/java/com/wynntils/features/user/ItemFavoriteFeature.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.google.common.reflect.TypeToken;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.TypeOverride;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.gui.render.RenderUtils;
+import com.wynntils.gui.render.Texture;
+import com.wynntils.mc.event.ContainerCloseEvent;
+import com.wynntils.mc.event.SlotRenderEvent;
+import com.wynntils.mc.utils.ComponentUtils;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wynn.item.GearItemStack;
+import com.wynntils.wynn.item.UnidentifiedItemStack;
+import com.wynntils.wynn.utils.ContainerUtils;
+import com.wynntils.wynn.utils.WynnUtils;
+import java.lang.reflect.Type;
+import java.util.HashSet;
+import java.util.Set;
+import net.minecraft.ChatFormatting;
+import net.minecraft.core.NonNullList;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class ItemFavoriteFeature extends UserFeature {
+    public static ItemFavoriteFeature INSTANCE;
+
+    @Config(visible = false)
+    public Set<String> favoriteItems = new HashSet<>();
+
+    @TypeOverride
+    private static final Type favoriteItemsType = new TypeToken<Set<String>>() {}.getType();
+
+    @SubscribeEvent
+    public void onChestCloseAttempt(ContainerCloseEvent.Pre e) {
+        if (!WynnUtils.onWorld()) return;
+        if (!ContainerUtils.isLootOrRewardChest(McUtils.mc().screen)) return;
+
+        NonNullList<ItemStack> items = ContainerUtils.getItems(McUtils.mc().screen);
+        for (int i = 0; i < 27; i++) {
+            ItemStack stack = items.get(i);
+
+            if (isFavorited(stack)) {
+                McUtils.sendMessageToClient(new TranslatableComponent("feature.wynntils.itemFavorite.closingBlocked")
+                        .withStyle(ChatFormatting.RED));
+                e.setCanceled(true);
+                return;
+            }
+        }
+    }
+
+    @SubscribeEvent
+    public void onRenderSlot(SlotRenderEvent.Post event) {
+        ItemStack itemStack = event.getSlot().getItem();
+
+        if (isFavorited(itemStack)) {
+            renderFavoriteItem(event);
```

Comment:
Could this be made a property?

Reply:
Lock is not a property either atm, Maybe it could be, but we still need the feature for blocking chest closing so

Reply:
Yeah, maybe this is better for a future pr.